src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,249 @@
+        public String toString() {
+            return "Node{"
+                    + "hash="
+                    + hash
+                    + ", key="
+                    + key
+                    + ", value="
+                    + value
+                    + ", next="
+                    + next
+                    + '}';
+        }
+    }
+
+    public void setValue(Node<K, V> node,V newValue) {
+        node.value = newValue;
+    }
+
+    public K getKey(Node<K, V> node) {
+        return node.key;
+    }
+
+    public V getValueFromNode(Node<K, V> node) {
+        return node.value;
+    }
+
+    public V[] getAllValues() {
+        V[] valuesArray = (V[]) new Object[size];
+        int index = 0;
+        for (Node<K, V> node : arrayNodes) {
+            if (node != null) {
+                valuesArray[index] = node.value;
+                index++;
+            }
+            if (node != null && node.next != null) {
+                Node<K, V> nextNode = node.next;
+                while (nextNode != null) {
+                    valuesArray[index] = nextNode.value;
+                    index++;
+                    nextNode = nextNode.next;
+                }
+            }
+            if (index >= size) {
+                break;
+            }
+        }
+        return valuesArray;
+    }
+
+    public K[] getAllKeys() {
+        K[] keysArray = (K[]) new Object[size];
+        int index = 0;
+        for (Node<K, V> node : arrayNodes) {
+            if (node != null) {
+                keysArray[index] = getKey(node);
+                index++;
+            }
+            if (node != null && node.next != null) {
+                Node<K, V> nextNode = node.next;
+                while (nextNode != null) {
+                    keysArray[index] = nextNode.key;
+                    index++;
+                    nextNode = nextNode.next;
+                }
+            }
+            if (index >= size) {
+                break;
+            }
+        }
+        return keysArray;
+    }
+
+    public void resize() {
+        int oldCapacity = arrayNodes.length;
+        size = 0;
+        Node<K, V>[] oldArray = arrayNodes;
+        arrayNodes = (Node<K, V>[]) new Node[oldCapacity * 2];
+        threshold = (int) (arrayNodes.length * loadFactor);
+        transfer(oldArray);
+    }
+
+    public void transfer(Node<K, V>[] oldArray) {
+        Node<K, V> currentNode = null;
+        for (Node<K, V> transferedNode: oldArray) {
+            if (transferedNode != null) {
+                put(getKey(transferedNode), getValueFromNode(transferedNode));
+                currentNode = transferedNode.next;
+            }
+            while (currentNode != null) {
+                put(getKey(currentNode), getValueFromNode(currentNode));
+                currentNode = currentNode.next;
+            }
+        }
+    }
+
+    public boolean isEmpty() {
+        return getSize() == 0;
+    }
+
+    public int getHash(Node<K, V> node) {
```

Comment:
```suggestion
    public int getHash(K key) {
```

I dont think we need whole node here